collect_app/src/androidTest/java/org/odk/collect/android/storage/StorageMigrationBannerTest.java

```diff
@@ -0,0 +1,36 @@
+package org.odk.collect.android.storage;
+
+import androidx.test.rule.ActivityTestRule;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.odk.collect.android.activities.MainMenuActivity;
+import org.odk.collect.android.support.StorageMigrationNotPerformedRule;
+import org.odk.collect.android.support.pages.MainMenuPage;
+
+public class StorageMigrationBannerTest {
+
+    @Rule
+    public ActivityTestRule<MainMenuActivity> main = new ActivityTestRule<>(MainMenuActivity.class);
+
+    @Rule
+    public StorageMigrationNotPerformedRule storageMigrationNotPerformedRule = new StorageMigrationNotPerformedRule();
+
+    @Test
+    public void when_storageMigrationNotPerformed_should_bannerBeVisible() {
+        new MainMenuPage(main)
+                .assertStorageMigrationBannerIsDisplayed()
+                .rotateToLandscape(new MainMenuPage(main))
+                .assertStorageMigrationBannerIsDisplayed()
+                .rotateToPortrait(new MainMenuPage(main))
+                .assertStorageMigrationBannerIsDisplayed()
+                .reopenApp()
+                .assertStorageMigrationBannerIsDisplayed();
+    }
+
+    @Test
+    public void when_learMoreButtonClicked_should_storageMigrationDialogAppear() {
+        new MainMenuPage(main)
```

Comment:
I think you'd be fine to combine this with the previous test as one flow. I think we can also move rotation/reopen testing out of Espresso for these cases but I'll have a quick think about that and get back to you.

Reply:
Yeah you should be able to use the `ActivityScenario` in [AndroidX testing](http://robolectric.org/androidx_test/) and just call `recreate` to test recreation (otation is really just that). We're using it for a fragment test in `FormMetadataFragmentTest`. We should be able to use this kind of pattern to test the `MainMenuActivity` directly rather than having to worry about rotation/recreation at the Espresso level.

Reply:
> think you'd be fine to combine this with the previous test as one flow.

Done.

when it comes to screen rotation I used recreate in some cases for example where I wanted to simulate that the app is [reopened](https://github.com/opendatakit/collect/pull/3623/files#diff-cb71153b5f29ea480658a710c445028bR85). However in this case rotation and recreation differ. In MainMenuActivity we use
`android:configChanges="locale|orientation|screenSize"`

that means `onDestroy` won't be called if you rotate your device so I expect a different behavior compared to recreation. For example if we have the banner with success msg it should disappear after reopening the app but should be still visible after simple rotation.